controllers/managedfusion_controller.go

```diff
@@ -689,6 +723,123 @@ func (r *ManagedFusionReconciler) reconcilePrometheusProxyNetworkPolicy() error
 	return nil
 }
 
+func (r *ManagedFusionReconciler) reconcileEgressFirewall() error {
+	if !r.AvailableCRDs[EgressFirewallCRD] {
+		r.Log.Info("EgressFirewall CRD not present, skipping reconcile for egress firewall")
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressFirewall, func() error {
+		if err := r.own(r.egressFirewall); err != nil {
+			return err
+		}
+		desiredEgressSpec, err := r.getEgressDesiredState()
+		if err != nil {
+			return err
+		}
+		r.egressFirewall.Spec = *desiredEgressSpec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update egressFirewall: %v", err)
+	}
+	return nil
+
+}
+
+func (r *ManagedFusionReconciler) reconcileEgressNetworkPolicy() error {
+	if !r.AvailableCRDs[EgressNetworkPolicyCRD] {
+		r.Log.Info("EgressNetworkPolicy CRD not present, skipping reconcile for egress network policy")
+		return nil
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressNetworkPolicy, func() error {
+		if err := r.own(r.egressNetworkPolicy); err != nil {
+			return err
+		}
+		desiredEgressFirewallSpec, err := r.getEgressDesiredState()
+		if err != nil {
+			return err
+		}
+		var desiredEgressNetworkPolicy openshiftv1.EgressNetworkPolicy
+		for i := range desiredEgressFirewallSpec.Egress {
+			desiredEgressNetworkPolicy.Spec.Egress = append(
+				desiredEgressNetworkPolicy.Spec.Egress,
+				openshiftv1.EgressNetworkPolicyRule{
+					Type: openshiftv1.EgressNetworkPolicyRuleType(desiredEgressFirewallSpec.Egress[i].Type),
+					To:   openshiftv1.EgressNetworkPolicyPeer(desiredEgressFirewallSpec.Egress[i].To),
+				},
+			)
+		}
+		r.egressNetworkPolicy.Spec = desiredEgressNetworkPolicy.Spec
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to update egressNetworkPolicy: %v", err)
+	}
+	return nil
+}
+
+func (r *ManagedFusionReconciler) getEgressDesiredState() (*ovnv1.EgressFirewallSpec, error) {
```

Comment:
- After looking through both CRDs why not return only the rules here?

``` go
type egressRules struct {
 type_ string
 dnsName string
 cidrSelector string
}

// callee signature be
func (r *ManagedFusionReconciler) getDesiredEgressRules() ([]egressRules, error)

// caller will just use either `dnsName` or `cidrSelector` whatever is set
```

(or)

- if having another struct is a no-go, let the callee fill the rules when the spec if supplied and switch based on available CRDs?

rationale being, the current implementation is trying to retrieve the info even though we were the one's who set the info in the first place.

Reply:
Can you check if the latest commit has the changes that you suggested?